app/decorators/case_contact_decorator.rb

```diff
@@ -100,4 +113,30 @@ def show_contact_type?(contact_type_id)
   def additional_expenses_count
     object.additional_expenses.any? ? object.additional_expenses.length : 0
   end
+
+  def address_of_volunteer
+    if volunteer_address && !volunteer_address.empty?
```

Comment:
Can this just be simplified to:

```suggestion
    if volunteer_address.present?
```

?

Reply:
It still needs a safe operator (can't guarantee any specific param is sent over by the form unlike single page forms), but otherwise yours is better.